src/main/java/xyz/thepathfinder/android/Cluster.java

```diff
@@ -244,31 +270,52 @@ public Cluster getSubcluster(String name) {
 
     /**
      * Returns an immutable collection of this cluster's direct subclusters.
+     *
      * @return An immutable collection of clusters.
      */
     public Collection<Cluster> getSubclusters() {
```

Comment:
I think its a little inconsistent that some of your methods return collections and others return lists. I'm guessing you're using collections because you like Collections#unmodifiableCollection.

If you want immutability, would you consider taking a dependency on guava (https://github.com/google/guava) and using ImmutableList? (http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/collect/ImmutableList.html)